engine/src/main/java/org/terasology/world/block/family/AttachedToSurfaceFamily.java

```diff
@@ -15,47 +15,85 @@
  */
 package org.terasology.world.block.family;
 
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
 import com.google.common.collect.Maps;
+import org.terasology.math.Rotation;
 import org.terasology.math.Side;
 import org.terasology.math.geom.Vector3i;
 import org.terasology.naming.Name;
+import org.terasology.registry.In;
 import org.terasology.world.BlockEntityRegistry;
 import org.terasology.world.WorldProvider;
 import org.terasology.world.block.Block;
+import org.terasology.world.block.BlockBuilderHelper;
 import org.terasology.world.block.BlockUri;
+import org.terasology.world.block.loader.BlockFamilyDefinition;
+import org.terasology.world.block.shapes.BlockShape;
 
 import java.util.Locale;
 import java.util.Map;
+import java.util.Set;
 
 /**
  */
+@RegisterBlockFamily("attachedToSurface")
+@BlockSections({"front", "left", "right", "back", "top", "bottom"})
```

Comment:
are block sections generic or what is the reason to handle them via plain strings?

Reply:
you can define your own sections in the family. This is carried over from the old implementation that used a factory.